eng/Build.props

```diff
@@ -117,6 +120,17 @@
         <ProjectToBuild Condition=" '$(BuildJava)' == 'true'" Include="@(JavaProjects)" Exclude="@(ProjectToExclude)" />
         <ProjectToExclude Condition=" '$(BuildJava)' != 'true'" Include="@(JavaProjects)" />
 
+        <!-- These projects have a transitive dependency on Microsoft.AspNetCore.Components.Web.JS exclude these projects if we are not building NodeJS -->
+        <!-- Do not skip these projects on the CI since we don't want to accidentally miss building artifacts -->
+        <DotnetProjectsTransitiveNodeJsDependencies Include="
+                          $(RepoRoot)src\Framework\**\*.csproj;
+                          $(RepoRoot)src\Analyzers\Analyzers\test\Microsoft.AspNetCore.Analyzers.Test.csproj;
+                          $(RepoRoot)src\ProjectTemplates\*\*.csproj;
+                          $(RepoRoot)src\ProjectTemplates\testassets\*\*.csproj;
+                          $(RepoRoot)src\Components\**\*.*proj;
+                          $(RepoRoot)src\Mvc\**\*.*proj;" />
+        <ProjectToExclude Include="@(DotnetProjectsTransitiveNodeJsDependencies)" Condition="'$(BuildNodeJS)' == 'false' and '$(ContinuousIntegrationBuild)' != 'true'" />
```

Comment:
This makes sense, but do we want to generate this set and store using something like: https://github.com/aspnet/AspNetCore/blob/master/eng/CodeGen.proj#L10

I guess it's not a huge difference because you still need to know that you need to do something.

Reply:
Could consider this as an improvement in the future.